src/client/src/models/Scene.js

```diff
@@ -112,20 +137,52 @@ const Scene = {
   // self-explanatory. Sets the currentScene property to the scene corresponding to the sceneId
   setCurrentScene: function(sceneId) {
     const list = Scene.getScenes()
-    Scene.currentScene = list[sceneId]
+    Scene.currentScene = list[Scene.findSceneIndex(sceneId)]
   },
 
   currentCaption: null,
 
   setCurrentCaption: function(captionId) {
-    Scene.currentCaption = Scene.currentScene.captions[captionId]
+    Scene.currentCaption = Scene.currentScene.captions[Scene.findCaptionIndex(captionId)]
+  },
+
+  findCaptionIndex: function (captionId) {
+    for (var i = 0; i < Scene.currentScene.captions.length; i++) {
+      if (Scene.currentScene.captions[i].id == captionId) {
+        return i
+      }
+    }
+
+    return -1
+  },
+
+  addCaption: function () {
+    var caption_index = 0
+    // Finds the max scene id in the list
+    if (Scene.currentScene.captions.length == 0) {
```

Comment:
is the if/else necessary if we can do `if (Scene.currentScene.captions.length > 0) { execute code }`. caption index is initialized to 0 on line 160

Reply:
Caption Index was not initialized. That was a naming error made by me, but the caption length should always return a non negative number.